compiler/modules/front/stmt.bal

```diff
@@ -276,7 +276,7 @@ function codeGenScope(StmtContext cx, bir:BasicBlock bb, Environment initialEnv,
             applyEffect(env, narrowings, effect);
             // Compound statements will gen narrowings post-block, no need to narrow after last stmt
             if curBlock != () && stmtIndex != lastStmtIndex {
-                addNarrowings(cx, curBlock, env, effect.narrowings, stmt.endPos);
+                addNarrowings(cx, curBlock, env, effect.narrowings, stmt.startPos);
```

Comment:
Not sure about this. See what @manuranga thinks.

Reply:
I don't think this is correct, @heshanpadmasiri does this break the validation? May I have a chat with you?

Reply:
Had a chat,
Option 1) We'll leave it as endPos, since it doesn't highlight nor should it be debugable via breakpoints
Option 2) Rotate the loop such that each iteration generate narrowings for the previous, then we can use the startPos of current statement as the narrowing pos of the previous.

I think let's go with Option 1.

Reply:
fixed

Reply:
It's not OK to use an endPos as the position of a BIR insn.

Reply:
If you what you want is the position of the close brace, then you should use that.

Reply:
We (potentially) add narrowing between each statement, so close brace is not a good fit. @heshanpadmasiri shall we explore above options 2 in that case?

(I have a minor concern with that though, if feels more correct for the narrowing to be part of the previous statement, therefore derive position from that)

Reply:
fixed